prebackup/prebackup.go

```diff
@@ -190,43 +227,76 @@ func (p *PreBackup) getClientset() (*kubernetes.Clientset, error) {
 func (p *PreBackup) waitForReady(watcher watch.Interface) error {
 	defer watcher.Stop()
 
-	for event := range watcher.ResultChan() {
-		deployment := event.Object.(*appsv1.Deployment)
-
-		switch event.Type {
-		case watch.Modified:
-
-			last := p.getLastDeploymentCondition(deployment)
-
-			if last != nil && isDeadlineExceeded(last) {
-				return fmt.Errorf("error starting pre backup pod %v: %v", deployment.GetName(), last.Message)
+	p.Log.V(2).Info("waiting on watcher ", "deployment", fmt.Sprintf("%s/%s", p.Obj.GetMetaObject().GetNamespace(), p.Obj.GetMetaObject().GetName()))
+	for {
+		select {
+		case event := <-watcher.ResultChan():
+			endWatch, err := p.handleEvent(event)
+			if err != nil {
+				return err
 			}
-
-			if hasAvailableReplica(deployment) {
+			if endWatch {
 				return nil
 			}
+		case <-p.CTX.Done():
+			p.Log.Error(p.CTX.Err(), "unexpected end during deployment watch ", "deployment", fmt.Sprintf("%s/%s", p.Obj.GetMetaObject().GetNamespace(), p.Obj.GetMetaObject().GetName()))
+			return p.CTX.Err()
+		}
+	}
+}
+
+func (p *PreBackup) handleEvent(event watch.Event) (bool, error) {
+	deployment, ok := event.Object.(*appsv1.Deployment)
+	if !ok {
+		p.Log.V(2).Info("unexpected event during deployment watch ", "event source", event.Object, "event type", event.Type)
```

Comment:
Why log level 2?  level 1 is already "debug", is "debug" enough to display?
Please mind that currently K8up can only log level 0 or 1